cmd/gateway-sia.go

```
@@ -0,0 +1,508 @@
+		return nil, err
+	}
+	return resp, nil
+}
+
+// post makes an API call and discards the response. An error is returned if
+// the response status is not 2xx.
+func post(addr, call, vals, apiPassword string) error {
+	resp, err := apiPost(addr, call, vals, apiPassword)
+	if err != nil {
+		return err
+	}
+	resp.Body.Close()
+	return nil
+}
+
+// list makes a lists all the uploaded files, decodes the json response.
+func list(addr string, apiPassword string, obj interface{}) error {
+	resp, err := apiGet(addr, "/renter/files", apiPassword)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNoContent {
+		return errors.New("Expecting a response, but API returned status code 204 No Content")
+	}
+
+	return json.NewDecoder(resp.Body).Decode(obj)
+}
+
+// get makes an API call and discards the response. An error is returned if the
+// responsee status is not 2xx.
+func get(addr, call, apiPassword string) error {
+	resp, err := apiGet(addr, call, apiPassword)
+	if err != nil {
+		return err
+	}
+	resp.Body.Close()
+	return nil
+}
+
+// newSiaGatewayLayer returns Sia gatewaylayer
+func newSiaGatewayLayer(host string) (GatewayLayer, error) {
+	sia := &siaObjects{
+		Address: host,
+		// RootDir uses access key directly, provides partitioning for
+		// concurrent users talking to same sia daemon.
+		RootDir:  os.Getenv("MINIO_ACCESS_KEY"),
+		TempDir:  os.Getenv("SIA_TEMP_DIR"),
+		password: os.Getenv("SIA_API_PASSWORD"),
+	}
+
+	// If Address not provided on command line or ENV, default to:
+	if sia.Address == "" {
+		sia.Address = "127.0.0.1:9980"
+	}
+
+	// If local Sia temp directory not specified, default to:
+	if sia.TempDir == "" {
+		sia.TempDir = ".sia_temp"
+	}
+
+	var err error
+	sia.TempDir, err = filepath.Abs(sia.TempDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the temp directory with proper permissions.
+	// Ignore error when dir already exists.
+	_ = os.Mkdir(sia.TempDir, 0700)
```

we should check if `os.IsExist(err)` is true before we ignore the error.

---

done using os.MkdirAll